packages/orchestration/src/util/filtering.test.ts

```diff
@@ -0,0 +1,214 @@
+      const filtering: FilteringModuleConfig = {
+        input: buildAzureContentFilter({
+          Hate: 4,
+          SelfHarm: 0,
+          Sexual: 2,
+          Violence: 6
+        }),
+        output: buildAzureContentFilter({ Sexual: 2, Violence: 6 })
+      };
+      const expectedFilterConfig: FilteringModuleConfig = {
+        input: {
+          filters: [
+            {
+              type: 'azure_content_safety',
+              config: {
+                Hate: 4,
+                SelfHarm: 0,
+                Sexual: 2,
+                Violence: 6
+              }
+            }
+          ]
+        },
+        output: {
+          filters: [
+            {
+              type: 'azure_content_safety',
+              config: {
+                Sexual: 2,
+                Violence: 6
+              }
+            }
+          ]
+        }
+      };
+      config.filtering = filtering;
+      const completionPostRequest: CompletionPostRequest =
+        constructCompletionPostRequest(config, prompt);
+      expect(
+        completionPostRequest.orchestration_config.module_configurations
+          .filtering_module_config
+      ).toEqual(expectedFilterConfig);
+    });
+
+    it('omits filters if not set', async () => {
+      const filtering: FilteringModuleConfig = {
+        input: buildAzureContentFilter(),
+        output: buildAzureContentFilter()
+      };
+      config.filtering = filtering;
+      const completionPostRequest: CompletionPostRequest =
+        constructCompletionPostRequest(config, prompt);
+      const expectedFilterConfig: FilteringModuleConfig = {
+        input: {
+          filters: [
+            {
+              type: 'azure_content_safety'
+            }
+          ]
+        },
+        output: {
+          filters: [
+            {
+              type: 'azure_content_safety'
+            }
+          ]
+        }
+      };
+      expect(
+        completionPostRequest.orchestration_config.module_configurations
+          .filtering_module_config
+      ).toEqual(expectedFilterConfig);
+    });
+
+    it('omits filter configuration if not set', async () => {
+      const filtering: FilteringModuleConfig = {};
+      config.filtering = filtering;
+      const completionPostRequest: CompletionPostRequest =
+        constructCompletionPostRequest(config, prompt);
+      expect(
+        completionPostRequest.orchestration_config.module_configurations
+          .filtering_module_config
+      ).toBeUndefined();
+    });
+
+    it('throw error when configuring empty filter', async () => {
+      const createFilterConfig = () => {
+        {
+          buildAzureContentFilter({});
+        }
+      };
+      expect(createFilterConfig).toThrow(
+        'Filter property cannot be an empty object'
+      );
+    });
+  });
+
+  describe('Azure content filter', () => {
+    it('builds filter config', async () => {
+      const filterConfig = buildAzureContentSafetyFilter({
```

Comment:
[req] I would first merge your PR into this. As we end up committing a new function with old logic.

Reply:
Done.